GitExtensions/Program.cs

```diff
@@ -57,6 +67,10 @@ private static void Main()
             // the JIT processing more types than required to configure NBug.
             // In this way, there's more chance we can handle startup exceptions correctly.
             RunApplication();
+
+            Logger.Info("Application stopped");
+            source.Cancel();
+            Logger.WaitShutdown();
```

Comment:
I think a more correct implementation would be to put these into a handler for `Application.ApplicationExit` event
https://docs.microsoft.com/en-us/dotnet/api/system.windows.forms.application.applicationexit?view=netframework-4.8

Reply:
...and verify it is getting invoked when app crashes with unhandled exceptions

Reply:
agree